hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/common/table/read/TestSpark35RecordPositionMetadataColumn.scala

```diff
@@ -69,81 +64,54 @@ class TestSpark35RecordPositionMetadataColumn extends SparkClientFunctionalTestH
 
   @Test
   def testRecordPositionColumn(): Unit = {
-    val _spark = spark
-    // Prepare the schema
-    val dataSchema = new StructType(
-      Array(
-        StructField("userid", IntegerType, nullable = false),
-        StructField("country", StringType, nullable = false),
-        StructField("ts", StringType, nullable = false)
+    //feature is only supported in spark 3.5+
+    if (HoodieSparkUtils.gteqSpark3_5) {
+      val _spark = spark
+      // Prepare the schema
+      val dataSchema = new StructType(
+        Array(
+          StructField("userid", IntegerType, nullable = false),
+          StructField("country", StringType, nullable = false),
+          StructField("ts", StringType, nullable = false)
+        )
       )
-    )
-    val requiredSchema = HoodieFileGroupReaderBasedParquetFileFormat
-      .getAppliedRequiredSchema(
-        dataSchema,
-        shouldUseRecordPosition = true,
```

Comment:
`shouldUseRecordPosition` (whether merging updates based on the positions encoded in the log block header) is independent of getting record positions from the Spark parquet reader with predicate pushdown.  Can we revert the changes in the test class?

Reply:
ok, fixed